src/header/mod.rs

```diff
@@ -320,10 +320,11 @@ impl Headers {
     }
 
     /// Removes a header from the map, if one existed.
-    /// Returns true if a header has been removed.
-    pub fn remove<H: Header>(&mut self) -> bool {
+    /// Returns the header, if one has been removed.
```

Comment:
Both feel yucky. I'd still prefer to just have `Option<H>`. The upgrade path could be to check `headers.has()` before calling `remove()`.